### PR TITLE
add a parameter to be able to configure context url of a webapp i.e archive name

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: Olivier Lamy <olamy@apache.org>

#### Changes proposed in this pull request:

-
-
-

**Fixes**: #

### CODE REVIEW

File: jetty-embedded-10/src/main/java/org/jboss/arquillian/container/jetty/embedded_10/ArquillianAppProvider.java [hunk 1]
```diff
@@ -96,10 +98,18 @@ protected App createApp(final Archive<?> archive) {
...
+                Path tmpDirectory = Files.createTempDirectory("arquillian-jetty");
+                Path archivePath = tmpDirectory.resolveSibling(archive.getName());
+                Files.deleteIfExists(archivePath);
+                exported = Files.createFile(archivePath).toFile();
+                exported.deleteOnExit();
+            } else {
+                // If this method returns successfully then it is guaranteed that:
+                // 1. The file denoted by the returned abstract pathname did not exist before this method was invoked, and
+                // 2. Neither this method nor any of its variants will return the same abstract pathname again in the current invocation of the virtual machine.
+                exported = File.createTempFile(EXPORT_FILE_PREFIX, archive.getName(), EXPORT_DIR);
```
Comment (joakime) [inline jetty-embedded-10/src/main/java/org/jboss/arquillian/container/jetty/embedded_10/ArquillianAppProvider.java:111]: It's still `archive.getName()` on the else branch too?

File: jetty-embedded-11/src/main/java/org/jboss/arquillian/container/jetty/embedded_11/ArquillianAppProvider.java [hunk 2]
```diff
@@ -96,10 +98,18 @@ protected App createApp(final Archive<?> archive) {
...
+                Path tmpDirectory = Files.createTempDirectory("arquillian-jetty");
+                Path archivePath = tmpDirectory.resolveSibling(archive.getName());
+                Files.deleteIfExists(archivePath);
+                exported = Files.createFile(archivePath).toFile();
+                exported.deleteOnExit();
+            } else {
+                // If this method returns successfully then it is guaranteed that:
+                // 1. The file denoted by the returned abstract pathname did not exist before this method was invoked, and
+                // 2. Neither this method nor any of its variants will return the same abstract pathname again in the current invocation of the virtual machine.
+                exported = File.createTempFile(EXPORT_FILE_PREFIX, archive.getName(), EXPORT_DIR);
```
Comment (joakime) [inline jetty-embedded-11/src/main/java/org/jboss/arquillian/container/jetty/embedded_11/ArquillianAppProvider.java:111]: Ditto, same else branch

File: jetty-embedded-10/src/main/java/org/jboss/arquillian/container/jetty/embedded_10/ArquillianAppProvider.java (hunk 1 above)
Comment (olamy) [reply jetty-embedded-10/src/main/java/org/jboss/arquillian/container/jetty/embedded_10/ArquillianAppProvider.java:111]: yes it is. Except the war file is created using only the archive.getName() in a tmp directory. 
So the context in Jetty will the archive.getName() (minus `.war`).

File: jetty-embedded-11/src/main/java/org/jboss/arquillian/container/jetty/embedded_11/ArquillianAppProvider.java (hunk 2 above)
Comment (olamy) [reply jetty-embedded-11/src/main/java/org/jboss/arquillian/container/jetty/embedded_11/ArquillianAppProvider.java:111]: I know there is a lot of code duplicated for each jetty 9/10/11 but I don't really want to touch this now.

**Review by joakime — APPROVED** [review]

**Pull request merged**